### PR TITLE
CPLAT-15339 Add Badge / Label migrators

### PR DESCRIPTION
**Pull request by joebingham-wk**
## Motivation
  
This PR adds the necessary migrators to convert WSD `Badge` and `Labels` to their corresponding MUI Chip variant.

## Changes
  

- Add the `BadgeToMuiChipMigrator` and `LabelToMuiChipMigrator`
- Extrapolate `HitAreaMixin` logic from the `MuiButtonMigrator` so it could be shared with the `Badge` migrator
  - Tests were rafactored to the same effect

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
      - [ ] CI is passing (and tests cover all necessary scenarios)
      - [ ] Run the codemod on a repo or two to verify it works as expected. I run it on xbrl-module and copy-ui.
          ```
          # activate the codemod (while located in the repo directory)
          pub global activate -s path .

          # Run the codemod (while located in the repo)
          pub global run over_react_codemod:mui_migration --component Label,Badge
          ```
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

File: lib/src/mui_suggestors/components/mui_button_migrator.dart
Comment (joebingham-wk) [inline lib/src/mui_suggestors/components/mui_button_migrator.dart:168]: These were moved into the shared `hit_area.dart` migrator mixin

File: lib/src/mui_suggestors/components/shared_props_migrators/hit_area.dart
Comment (joebingham-wk) [inline lib/src/mui_suggestors/components/shared_props_migrators/hit_area.dart:7]: All of the logic in this file are pulled from the `MuiButtonMigrator` and is completely unchanged

File: test/mui_suggestors/components/shared_props_migrators/hit_area.dart
Comment (joebingham-wk) [inline test/mui_suggestors/components/shared_props_migrators/hit_area.dart:32]: These tests were mostly pulled from the button migrator test suite and then refactored a little to change the factory name and such

File: lib/src/mui_suggestors/components/shared_props_migrators/hit_area.dart
Comment (aaronlademann-wf) [reply lib/src/mui_suggestors/components/shared_props_migrators/hit_area.dart:23]: Is it not possible to share them across migrators?

File: lib/src/mui_suggestors/components/chip/badge_to_mui_chip_migrator.dart
Comment (aaronlademann-wf) [inline lib/src/mui_suggestors/components/chip/badge_to_mui_chip_migrator.dart:58]: Wouldn't `href` need something similar? Or is `href` a prop on a MUI `Chip`?

File: lib/src/mui_suggestors/components/chip/badge_to_mui_chip_migrator.dart
Comment (aaronlademann-wf) [inline lib/src/mui_suggestors/components/chip/badge_to_mui_chip_migrator.dart:109]: Are both of these supposed to suggest `'mr': 0`?

File: lib/src/mui_suggestors/components/shared_props_migrators/color_props.dart
Comment (aaronlademann-wf) [inline lib/src/mui_suggestors/components/shared_props_migrators/color_props.dart:105]: #nit I'm not really certain what the purpose of `palette.common.white` is in MUI - but I think its probably safe to just do `'color': '#fff'` for the sake of brevity.

File: test/mui_suggestors/components/chip/badge_to_mui_chip_migrator_test.dart
Comment (aaronlademann-wf) [inline test/mui_suggestors/components/chip/badge_to_mui_chip_migrator_test.dart:62]:
```suggestion
              // Shadows the WSD Badge
```

**Review by aaronlademann-wf — COMMENTED** [review]
This looks really good @joebingham-wk! Should we try running it in a sourcegraph batch preview to see what the changes look like IRL?

File: lib/src/mui_suggestors/components/chip/badge_to_mui_chip_migrator.dart
Comment (joebingham-wk) [reply lib/src/mui_suggestors/components/chip/badge_to_mui_chip_migrator.dart:58]: Ope yep, we do need a similar handling for `href`. Good catch!

File: lib/src/mui_suggestors/components/chip/badge_to_mui_chip_migrator.dart
Comment (joebingham-wk) [reply lib/src/mui_suggestors/components/chip/badge_to_mui_chip_migrator.dart:109]: That was intentional, though I could be incorrect in my thinking in the case of PULL_LEFT. My thought was that we wouldn't want the additional padding because the use cases for pulling left meant that the badge wouldn't be placed directly next to the content (or why not just use LEFT or the default?). I may be missing a use case there though

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/mui_suggestors/components/chip/badge_to_mui_chip_migrator.dart
Comment (aaronlademann-wf) [reply lib/src/mui_suggestors/components/chip/badge_to_mui_chip_migrator.dart:109]: You might be right...

Comment (aaronlademann-wf) [reply lib/src/mui_suggestors/components/chip/badge_to_mui_chip_migrator.dart:109]: Either way its definitely not worth holding up the PR.

**Review by aaronlademann-wf — CHANGES_REQUESTED** [review]
@joebingham-wk one thing I just thought of is - for the FIXME comment when the `align` prop is found on a WSD usage, should we recommend wrapping a `Stack` around the badge and any adjacent content instead of using `sx` on the badge?

**Comment by joebingham-wk** [comment]
We could do that @aaronlademann-wf! That's certainly a more robust way to handle layout. Is there a way to easily get the floating behavior that `PULL_RIGHT` (or left) created? From playing around with `Stack`, it seems like a nice flexbox abstraction but doesn't give the same floating behavior. That may be fine, but then it's a little bit more of a refactor

**Comment by joebingham-wk** [comment]
Converted to draft because this needs to be refactored based on changes in https://github.com/Workiva/react_material_ui/pull/158.

**Comment by robbecker-wf** [comment]
closing as stale

**Pull request closed without merging**
